packages/circuits/src/initial.circom

```diff
@@ -0,0 +1,89 @@
+pragma circom 2.0.0;
+
+include "./merkle.circom";
+include "../node_modules/circomlib/circuits/comparators.circom";
+
+template CheckReplaceArray(n) {
+    signal input index;
+    signal input newValue;
+    signal input array[n];
+    signal output newArray[n];
+    signal output replacedValue;
+
+    component checkIndex[n];
+    signal items[n];
+    for (var i = 0; i < n; i++) {
+        checkIndex[i] = IsEqual();
+        checkIndex[i].in[0] <== i;
+        checkIndex[i].in[1] <== index;
+        newArray[i] <== (newValue - array[i]) * checkIndex[i].out + array[i];
+        // newArray[i] <== (checkIndex[i].out * newValue) + ((1 - checkIndex[i].out) * array[i]);
+        items[i] <== checkIndex[i].out * array[i];
+    }
+
+    // get the replaced value
+    var temp = 0;
+    for (var i = 0; i < n; i++) {
+        temp += items[i];
+    }
+    replacedValue <== temp;
+}
+
+template Initial(levels, cardCount) {
+    signal input deckRoot;
+    signal input newDeckRoot;
+    signal input deckLeaves[2**levels];
+    signal input initialDeckTailCardIndex;
+    signal input handRoot;
+    signal input newHandRoot;
+    signal input drawnCardIndices[cardCount];
+
+    component checkInitialDeck = CheckMerkleRoot(levels);
+    checkInitialDeck.root <== deckRoot;
+    checkInitialDeck.leaves <== deckLeaves;
+
+    // check initial hand is all null
+    var handLeaves[2**levels];
+    for (var i = 0; i < 2**levels; i++) {
+        handLeaves[i] = 255;
+    }
+    component checkInitialHand = CheckMerkleRoot(levels);
+    checkInitialHand.root <== handRoot;
+    checkInitialHand.leaves <== handLeaves;
```

Comment:
I don't think we need this check (that the initial hand is all null) — the hand will be initialized when the game is joined, and the hand is never written before. I will in fact *not* be equal to a root of all nulls, but instead to 0.

We can simply skip the `handRoot` signal altogether.

Reply:
That makes sense, will make the change!